input/postgres/statements.go

```diff
@@ -234,6 +236,22 @@ func GetStatements(server *state.Server, logger *util.Logger, db *sql.DB, global
 		if showtext {
 			statementTexts[key] = receivedQuery.String
 		}
+		if queryID.Valid && showtext {
+			server.QueryIdentitiesMutex.Lock()
```

Comment:
Two thoughts:

1) Should we be holding the mutex for the full loop, not each iteration? (the trade-off here is that we block other users for longer, but are reducing the per-loop overhead slightly)

2) If we are holding it separately for each iteration, should we first acquire the mutex for reading, and only elevate the mutex to writing if we didn't find the query ID? (that would be best in terms of letting others read too, but I'm not sure if that's necessary / helpful)

Reply:
This `rows.Next()` loop takes at least 400ms per the screenshot in #346. It's hard to say which approach is better, I had just chosen this option to avoid moving `Lock()` to line 209 and `Unlock` to line 260.

Using a read-only lock doesn't make sense here because `LastSeen` is always updated.